servant-auth-server/src/Servant/Auth/Server/Internal.hs

```diff
@@ -44,27 +46,28 @@ instance ( HasServer (AddSetCookieApi api) ctxs, AreAuths auths ctxs v
                   Secure -> True
                   NotSecure -> False
              }
-        cookies <- makeCookies authResult
-        return (authResult, csrf : cookies )
+        mCookies <- liftIO $ makeCookies authResult
+        case mCookies of
+          Just cookies -> return (authResult, csrf `SCCons` cookies)
+          Nothing -> delayedFailFatal err500
```

Comment:
This doesn't seem right. One way in which we get a `Nothing` is if certain JWT params were incorrectly picked, in which case 500 makes sense. But another is if the user is not authenticated. We don't want to 500 then - we want to let the developer pick the appropriate behavior. 

This is perhaps the crucial test to this approach. Previously cookies were only sent for authenticated users. To have that behavior here seems difficult, since the types would differ by branch.

One option, I think, is to have `Maybe`s in the HList, and use the new `noHeader` in `addSetCookies` when we see a `Nothing`. Then here one could simply have `SCCons Nothing cookies`.